### ISSUE TITLE
tracks on other paths than streets now showing up on the map

### ISSUE THREAD

**Issue by joshinils** [kartaview/openstreetcam.org#324]
awhen i look at https://kartaview.org/map/ it appears as if the blue lines only show up for streets, not paths or otherwise accessible ways (for bicycles, at least)
for example: https://kartaview.org/details/3580373/2172
only shows up when i zoom in far enough for the green dots to appear.
I would like for every track to be shown like the green dots, even at lower zoom levels, especially at higher zoom levels.

**Comment by joshinils** [comment]
For example: https://kartaview.org/details/3582857/10787/track-info 
There, i rode on https://www.openstreetmap.org/way/41211940 which is further south than the highlited franziskastraße above it.

Or does this have to do with me passing both the franziskastraße and utestraße at https://kartaview.org/details/3582857/11798/track-info ? 
and since https://www.openstreetmap.org/way/59454315 is a path it does not show up in blue

**Comment by nakic** [comment]
I have the same problem. I'm systematically mapping pedestrian paths, riverbanks, playgrounds etc. as well as roads, and this not being able to see the images on the main map is a significant issue. I'm promoting the imagery in a community of ~100 people in local councils and not being able to see imagery taken further away from streets may be the obstacle that just might make it too much for them too use.

**Comment by joshinils** [comment]
yeah, zoom 18 and higher show the green dots at the exact gps location of the images. not really useful at all

**Comment by brmonaghan** [comment]
I agree with this too! I have multiple tracks from KV that I cannot see without going to the right. Spinning error also occurs too with clicking on a track and spinning cursor appears for a few seconds and goes away. Also, tracks need to be on the path. @zhaitianduo any word on when this will be resolved? 🤔

**Comment by zhangtiedong0912** [comment]
@joshinils  hi let me explain: the green dots mean : the photo not matched the road . we don't support show gps point now.

**Comment by zhangtiedong0912** [comment]
@joshinils  thanks for you feedback.  KartaView will use original GPS info match road network. you will see the blur line (Actually the purple) if match success. you will see the green dots if match failed. So the blue lines only show up for streets.

**Comment by zhangtiedong0912** [comment]
We will redesign in api V2, expectations can lead to a better experience.

**Comment by nakic** [comment]
@zhangtiedong0912 Thanks for the clarification.

This looks a bit better than before: I don't think I could see the green dots at zoom > 18 before, so I imagine you made improvements there: thanks!

It has glitches, though. For example, I recorded a [track along a stream bank](https://kartaview.org/map/@45.792882237057285,15.936828644945876,19z) which seems to abruptly end at the north end, when it's clear from track photos that imagery is available for another few hundred meters north. I imagine this has to do with snapping images to a nearby road when one is close. I understand the value in snapping to roads, but it'd be useful to either refine the snapping algorithm or to have an option to avoid snapping.

Finally, it may make sense to show off-street images shown at lower zoom levels as well, perhaps using a different colour, depending. It'd sure be useful for my use case (using imagery to support local council work), but there might be others interested in this, as well.

**Comment by zhangtiedong0912** [comment]
@nakic hi your guess is right. Now just use gps do match work. Improving the match algorithm has always been a challenge
about show off-street images, Thanks for you suggestion , i will discuss with our team.

**Comment by nakic** [comment]
Thanks for the quick reply!

Just to add another suggestion that might be of help, perhaps a pragmatic solution might be to allow users who upload tracks to label a track as "on-street" or "off-street", with "on-street" as the default (currently hard-coded), so that pure GPS is used for off-street imagery. (In general, being able to edit track metadata (e.g. name) would be lovely, but that's another story. )

**Comment by zhangtiedong0912** [comment]
@nakic  i think you mean this . 
<img width="662" alt="image" src="https://user-images.githubusercontent.com/51616101/158133958-9e3e8ab6-2e64-4264-b7aa-1ca12593d3a2.png">

**Comment by nakic** [comment]
Yes, something like that, but that's only in the mini-map.

My impression is that the user uploading the imagery knows how the data should be viewed and that if uploader says "this is off-street, use GPS only", than that would be the way the imagery would be shown in the _main_ map.

Alternatively, if there was a "use original location" control in the _main_ map, that would be a good option, as well.

I say all this knowing full well that you have a much clearer idea of what your users ask for: I can only assume that and talk about what would be useful for users I'd like to bring to KartaView.

**Comment by zhangtiedong0912** [comment]
i see , thanks again

**Comment by joshinils** [comment]
hmm, so bike-tours taken to make images which include streets and forest-paths would need to be split and each uploaded separately? or would the detection be based on what kind of path is in the vicinity? would it be possible or helpful to tell kartaview which type of vehicle was used when taking the images? like a car or bicycle (which then may include paths accessible to bikes)?

**Comment by zhangtiedong0912** [comment]
@joshinils hi ，this feature is not currently supported. It is still the general processing logic
